### PR TITLE
Remove `ParametricPulse`

### PR DESCRIPTION
**Pull request by TsafrirA**
### Summary
This PR removes the `ParametericPulse` base class and pulse library.

### Details and comments
The `ParametricPulse` library has been superseded by the `SymbolicPulse` library which provides QPY support and some performance improvements. It is supposed to be deprecated in Qiskit 0.46 in #11279, and has been pending deprecation since Terra 0.22. This PR removes the base class and pulse library.

### CODE REVIEW

File: qiskit/qobj/converters/pulse_instruction.py [hunk 1]
```diff
@@ -50,12 +50,12 @@ class ParametricPulseShapes(Enum):
     @classmethod
     def from_instance(
         cls,
-        instance: Union[library.ParametricPulse, library.SymbolicPulse],
+        instance: library.SymbolicPulse,
     ) -> "ParametricPulseShapes":
         """Get Qobj name from the pulse class instance.
 
         Args:
-            instance: Symbolic or ParametricPulse class.
```
Comment (wshanks) [inline qiskit/qobj/converters/pulse_instruction.py:58]: The references to `library.parametric_pulses` below should be removed.

**Review by wshanks — COMMENTED** [review]
This looks fine but note that #11024 already does this. I put a comment in-line based on something I saw there but not here.

File: qiskit/qobj/converters/pulse_instruction.py (hunk 1 above)
Comment (TsafrirA) [reply qiskit/qobj/converters/pulse_instruction.py:58]: Thanks.

**Comment by TsafrirA** [comment]
@wshanks @nkanazawa1989 
Thanks @wshanks for the heads up. I forgot about #11024. It came first, and it's better to promote community contribution, so I am absolutely fine with going with that PR.

File: qiskit/visualization/pulse/matplotlib.py [hunk 2]
```diff
@@ -257,8 +256,6 @@ def _build_waveform(self):
             for command in commands:
                 duration = command.duration
                 tf = min(time + duration, self.tf)
-                if isinstance(command, ParametricPulse):
```
Comment (nkanazawa1989) [inline qiskit/visualization/pulse/matplotlib.py:260]: Do we need to replace this with `SymbolicPulse`?

File: qiskit/visualization/pulse_v2/layouts.py [hunk 3]
```diff
@@ -361,9 +361,7 @@ def time_map_in_ns(
     )
 
 
-def detail_title(
-    program: Union[pulse.Waveform, pulse.ParametricPulse, pulse.Schedule], device: DrawerBackendInfo
-) -> str:
+def detail_title(program: Union[pulse.Waveform, pulse.Schedule], device: DrawerBackendInfo) -> str:
```
Comment (nkanazawa1989) [inline qiskit/visualization/pulse_v2/layouts.py:364]: Same here

File: releasenotes/notes/remove-parametric-pulse-740cf905e96d6b82.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+upgrade:
+  - |
+    The ``ParametricPulse`` base class and pulse library were removed (previously located
+    at ``qiskit.pulse.library.parametric_pulses``), following their deprecation on
+    Qiskit 0.46.
+
+    The ``ParametricPulse`` base class and pulse library were superseded by the
+    :class:`~qiskit.pulse.SymbolicPulse` class and pulse library, which provide
+    better performance and QPY support. All removed library pulses have a
```
Comment (nkanazawa1989) [inline releasenotes/notes/remove-parametric-pulse-740cf905e96d6b82.yaml:10]: I'm not sure if symbolic will provide better performance. Visualization is usually slow due to symbolic expression evaluation overhead.

**Comment by TsafrirA** [comment]
@nkanazawa1989 following my previous comment, do you want to handle this here or on #11024?

**Comment by nkanazawa1989** [comment]
Close in favor of #11024 (can be reopened if another PR cannot meet 1.0 deadline)

**Pull request closed without merging**